dataplanes/vpp/pkg/nsmdataplane/nsmdataplane_server.go

```diff
@@ -48,22 +45,32 @@ type Update struct {
 	localMechanisms  []*common.LocalMechanism
 }
 
+var mechanisms = map[common.LocalMechanismType]nsmvpp.Mechanism{
+	common.LocalMechanismType_KERNEL_INTERFACE: nsmvpp.KernelInterface{},
+	common.LocalMechanismType_MEM_INTERFACE:    nsmvpp.MemifInterface{},
+	common.LocalMechanismType_VHOST_INTERFACE:  nsmvpp.UnimplementedMechanism{},
+	common.LocalMechanismType_SRIOV_INTERFACE:  nsmvpp.UnimplementedMechanism{},
+	common.LocalMechanismType_HW_INTERFACE:     nsmvpp.UnimplementedMechanism{},
+}
+
```

Comment:
Is this from PR #357 ?  If so, perhaps we should get that merged and rebase on top of it :)